cmd/query/app/apiv3/grpc_gateway_test.go

```diff
@@ -144,36 +222,100 @@ func testGRPCGatewayWithTenancy(
 	setupRequest func(*http.Request),
 ) {
 	gw := setupGRPCGateway(t, basePath, serverTLS, clientTLS, tenancyOptions)
+	t.Run("GetServices", func(t *testing.T) {
+		runGatewayGetServices(t, gw, setupRequest)
+	})
+	t.Run("GetOperations", func(t *testing.T) {
+		runGatewayGetOperations(t, gw, setupRequest)
+	})
+	t.Run("GetTrace", func(t *testing.T) {
+		runGatewayGetTrace(t, gw, setupRequest)
+	})
+	t.Run("FindTraces", func(t *testing.T) {
+		runGatewayFindTraces(t, gw, setupRequest)
+	})
+}
```

Comment:
@albertteoh this was the point of cleaning up tests - how they are testing all gateway APIs and we can try things like upgrading the gateway dep to v2.x or completely re-implementing it. My next step would be to try an upgrade, which requires building a new protoc image that I just kicked off (takes 4hrs to build), and maybe with v2 gateway we could find a way to integrate gogo marshaling.